src/erlcloud_aws.erl

```diff
@@ -494,3 +501,189 @@ get_filtered_statuses(ServiceNames, Statuses) ->
                 ServiceNames)
         end,
         Statuses).
+
+
+%%%---------------------------------------------------------------------------
+-spec profile() -> {ok, aws_config()} | {error, string()}.
+%%%---------------------------------------------------------------------------
+%% @doc Retrieve a config based on default credentials
+%%
+%% This function will retrieve the credentials for the <em>default</em>
+%% profile, in the same way as the AWS CLI tools, and construct a {@link
+%% #aws_config{}. config record} to be used accessing services.
+%%
+%% @see profile/2
+%%
+profile() ->
+    profile( default, [] ).
+
+
+%%%---------------------------------------------------------------------------
+-spec profile( Name :: atom() ) -> {ok, aws_config()} | {error, string()}.
+%%%---------------------------------------------------------------------------
+%% @doc Retrieve a config based on named credentials profile
+%%
+%% This function will retrieve the credentials for the named profile, in the
+%% same way as the AWS CLI tools, and construct a {@link
+%% #aws_config{}. config record} to be used accessing services.
+%%
+%% @see profile/2
+%%
+profile( Name ) ->
+    profile( Name, [] ).
+
+
+-type profile_option() :: {role_session_name, string()}
+                          | {role_session_secs, 900..3600}.
+
+%%%---------------------------------------------------------------------------
+-spec profile( Name :: atom(), Options :: [profile_option()] ) ->
+                     {ok, aws_config()} | {error, string()}.
+%%%---------------------------------------------------------------------------
+%% @doc Retrieve a config based on named credentials profile
+%%
+%% This function will read the <code>$HOME/.aws/credentials</code> file used
+%% by the AWS CLI tools and construct a {@link #aws_config{}. config record}
+%% to be used accessing services for the named profile.  This supports both
+%% direct credentials that appear in the profile:
+%%
+%% <code><pre>
+%%  [default]
+%%  aws_access_key_id = XXXXXXXXXXXXXXXXXXX2
+%%  aws_secret_access_key = yyyyyyyyyyyyyyyyyyyyyyyyyy+yyyy/yyyyyyyy2
+%% </pre></code>
+%%
+%% as well as indirection using <em>source_profile</em>:
+%%
+%% <code><pre>
+%%  [foo]
+%%  source_profile = default
+%% </pre></code>
+%%
+%% and finally, will supports the <em>role_arn</em> specification, and will
```

Comment:
I can add the expiration time to the `#aws_config{}` record so that callers can refer to it.

@motobob On the rest, I think what you are really asking for is to have `erlcloud_sts:assume_role` produce a config that will auto-renew? I wouldn't think that this should apply only to profile based configs, but all such configs.  If you agree, I'll be happy to add that support to the `erlcloud_sts` and the auto-renew support to `erlcloud_aws`.

BTW, this generally applies to metadata based credentials, which probably would benefit from having a single mechanism used for renewal, since there would be at least two different config credential generators at that point.